x-pack/plugins/xpack_main/public/services/path.js

```diff
@@ -6,11 +6,9 @@
 
 import chrome from 'ui/chrome';
 
-export function PathProvider($window) {
-  const path = chrome.removeBasePath($window.location.pathname);
-  return {
-    isUnauthenticated() {
-      return path === '/login' || path === '/logout' || path === '/logged_out' || path === '/status';
-    }
-  };
-}
+export const Path = {
+  isUnauthenticated() {
+    const path = chrome.removeBasePath(window.location.pathname);
+    return path === '/login' || path === '/logout' || path === '/logged_out' || path === '/status';
```

Comment:
Do we need the function to be wrapped in an object? Maybe

```js
const someRandomPaths = ['/login', '/logout', '/logged_out', '/status'];
export const isAuthenticated = () => {
  const path = chrome.removeBasePath(window.location.pathname);
  return someRandomPaths.indexOf(path) > -1;
};
```

I know you are refactoring here, but whoever wrote this before, seems very hacky to check if user is authenticating by checking `window.location`.

Reply:
🤷‍♂